omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,240 @@
+        assertEquals(generatedIdentifierSource.getIdentifierType().getUuid(), IDENTIFIER_TYPE_UUID);
+        assertEquals(initialIdentifierSourceCount + 1, getAllCount());
+
+    }
+
+    @Test
+    public void shouldSaveARemoteIdentifierGenerator() throws Exception {
+
+        long initialIdentifierSourceCount = getAllCount();
+
+        SimpleObject identifierSource = new SimpleObject();
+        identifierSource.add("name", "test remote identifier");
+        identifierSource.add("description", "test remote identifier");
+        identifierSource.add("url", "http://localhost");
+        identifierSource.add("username", "openmrsuser");
+        identifierSource.add("password", "openmrspassword");
+        identifierSource.add("identifierType", IDENTIFIER_TYPE_UUID);
+        identifierSource.add("sourceType", REMOTE_IDENTIFIER_SOURCE_TYPE);
+
+        String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
+
+        MockHttpServletRequest req = newPostRequest(getURI(), jsonIdentifierSource);
+
+        SimpleObject newIdentifierSource = deserialize(handle(req));
+
+        RemoteIdentifierSource generatedIdentifierSource = (RemoteIdentifierSource) service.getAllIdentifierSources(false).get(7);
+
+        assertEquals(PropertyUtils.getProperty(newIdentifierSource, "uuid"), generatedIdentifierSource.getUuid());
+        assertEquals(generatedIdentifierSource.getName(), "test remote identifier");
+        assertEquals(generatedIdentifierSource.getDescription(), "test remote identifier");
+        assertEquals(generatedIdentifierSource.getUrl(), "http://localhost");
+        assertEquals(generatedIdentifierSource.getUser(), "openmrsuser");
+        assertEquals(generatedIdentifierSource.getPassword(), "openmrspassword");
+        assertEquals(generatedIdentifierSource.getIdentifierType().getUuid(), IDENTIFIER_TYPE_UUID);
+        assertEquals(initialIdentifierSourceCount + 1, getAllCount());
+
+    }
+
+    @Test
+    public void shouldSaveAnIdentifierPool() throws Exception {
+
+        long initialIdentifierSourceCount = getAllCount();
+
+        SimpleObject identifierSource = new SimpleObject();
+        identifierSource.add("name", "test identifier pool");
+        identifierSource.add("description", "test identifier pool");
+        identifierSource.add("batchSize", "500");
+        identifierSource.add("minPoolSize", "5");
+        identifierSource.add("sourceUuid", getUuid());
+        identifierSource.add("identifierType", IDENTIFIER_TYPE_UUID);
+        identifierSource.add("sourceType", IDENTIFIER_POOL_SOURCE_TYPE);
+
+        String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
+
+        MockHttpServletRequest req = newPostRequest(getURI(), jsonIdentifierSource);
+
+        SimpleObject newIdentifierSource = deserialize(handle(req));
+
+        IdentifierPool generatedIdentifierSource = (IdentifierPool) service.getAllIdentifierSources(false).get(7);
```

Comment:
Why do you fetch all identifier sources instead of the one you are interested in?

Reply:
True. Replaced with: `IdentifierPool generatedIdentifierSource = (IdentifierPool) service.getIdentifierSourceByUuid(newIdentifierSourceUuid);`